### PR TITLE
WIP - Update Apache Arrow ADBC to 0.8.0

### PR DESCRIPTION
**Pull request by Cowa**
I need one of the new features from the version `0.8.0` so I tried to update it myself.

It's not tested as I can't get it to compile:
Capture d’écran 2023-11-24 à 10 53 13

`checksum.exs` seems to be required but is not versioned so I tried to remove it from `package()` but it still shows the error.

Any idea how to fix it? Thanks!

### CODE REVIEW

File: update.exs [hunk 1]
```diff
@@ -91,7 +91,7 @@ case String.split(File.read!(file), "# == CONSTANTS ==") do
     # == CONSTANTS ==
     """
 
-    File.write!(file, Code.format_string!(pre <> mid <> post) <> "\n")
+    File.write!(file, "#{Code.format_string!(pre <> mid <> post)}\n")
```
Comment (Cowa) [inline update.exs:94]: This was causing an issue:

```
(ArgumentError) construction of binary failed: segment 1 of type 'binary': expected a binary but got: ["defmodule", " ", "Adbc", ".Driver", " do", "\n  ", ...]
```

**Comment by cocoa-xu** [comment]
Hi @Cowa! Thank you very much for the PR.

> checksum.exs seems to be required but is not versioned so I tried to remove it from package() but it still shows the error.
>
> Any idea how to fix it? Thanks!

It's okay that `checksum.exs` is missing because it will automatically fallback to compiling from source. The issue in your system was that there was no `cmake` installed on the system. Sicne this project will use cmake to build C++ source files, this will lead to a fatal error. Install cmake should fix this issue. :)

**Comment by Cowa** [comment]
Thank you, I was missing cmake :)

In my branch when I run `make` it works.
But somehow in Livebook the compilation still fails with the same error 🤷‍♂️

**Comment by josevalim** [comment]
IF you are using Livebook via the desktop app, it does not have access to your PATH. You need to configure the PATH in settings. :)

**Comment by josevalim** [comment]
Thank you @Cowa! I have updated it locally. The reason I did so is because, for external contributions, I always need to review all code to ensure there was no modifications, so in this case it is easier for me to do the update myself. Feel free to ping us next time. :)

**Comment by Cowa** [comment]
No problem I understand, thank you both of you for your help and your reactivity! :)

**Comment by josevalim** [comment]
v0.2.2 is out!

**Pull request closed without merging**